### PR TITLE
fix(argo-cd): Fixed default tls hosts for grpc ingress endpoint

### PR DESCRIPTION
**Pull request by groundhog2k**
The default tls hosts entry for ingress must match the default ingress rules host - which can be .Values.server.ingressGrpc.hostname or  $hostname  when nothing else was specified.

Example to reproduce this bug:

Create a server with 2 ingresses (for argocd ui and grpc):
```
global:
  domain: "argocd.k8s.local"

server:
  ingress:
    enabled: true

    annotations:
      cert-manager.io/cluster-issuer: blablabla

    tls: true
    hostname: "argocd.k8s.local"

  ingressGrpc:
    enabled: true
    annotations:
      cert-manager.io/cluster-issuer: blablabla
      nginx.ingress.kubernetes.io/backend-protocol: "GRPC"

    tls: true
    hostname: "argocd-grpc.k8s.local"
```
The Argo CD UI ingress endpoint is correct.

The GRPC ingress endpoint will be rendered with wrong tls hosts path (grpc.argocd.k8s.local
 instead of argocd-grpc.k8s.local).

The PR will fix this problem.

---

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [ ] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [ ] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ appVersion: v2.10.2
 kubeVersion: ">=1.23.0-0"
 description: A Helm chart for Argo CD, a declarative, GitOps continuous delivery tool for Kubernetes.
 name: argo-cd
-version: 6.6.0
+version: 6.6.1
```
[the remark this answers is not in the document]
Comment (groundhog2k) [reply charts/argo-cd/Chart.yaml:6]: You mean the one in README.md ? -- Done.

[the remark this answers is not in the document]
Comment (yu-croco) [reply charts/argo-cd/Chart.yaml:6]: Not on README but on Chart.yaml. Please remove old one and add new one for this PR.
https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog

Required items are shown on PR description but you removed.
-> https://github.com/argoproj/argo-helm/blob/main/.github/pull_request_template.md

**Comment by mkilchhofer** [comment]
@pdrastil can I have your eyes on this as you refactored the whole Ingress piece? :-)

**Review by mbevc1 — APPROVED** [review]

**Review by pdrastil — APPROVED** [review]

**Comment by pdrastil** [comment]
LGTM - thanks for extra fallback

**Pull request merged**